packages/model-viewer/src/model-viewer.ts

```diff
@@ -34,7 +34,9 @@ export const ModelViewerElement = MagicLeapMixin(AnnotationMixin(StagingMixin(
 
 export type ModelViewerElement = InstanceType<typeof ModelViewerElement>;
 
-customElements.define('model-viewer', ModelViewerElement);
+if (!customElements.get('model-viewer')) {
```

Comment:
This seems like a case where we should be warning that `model-viewer` has already been defined. In fact, this change hides a error from them that would have warned them.

`model-viewer.js` is a gigantic bundle, so we should make the user aware that it is being loaded without having been defined.

Reply:
This was primarily to resolve a hot-reloading issue in a module setup. Works on the assumption that `model-viewer` would only have been defined as a `<ModelViewerElement>`.

The current error is a bit opaque. Would it be worth type checking the already-defined element and warning specifically if there's a mismatch / the library has been double-loaded? Either way I'll move this for discussion elsewhere.